### PR TITLE
PF-471 implement aqua calc5000 field data plugin

### PR DESCRIPTION
**Pull request by BillChen-AI**
@DougSchmidt-AI ,
  This is my first cut of the implementation. The 22 samples files can all be parsed and imported. 
  Can you take a look? Especially at some of the mapping logic and see if they reasonable?

### CODE REVIEW

File: src/AquaCalc5000/Parsers/HeaderParser.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        public HeaderParser(CsvParser csvParser)
+        {
+            _csvParser = csvParser;
+        }
+
+        public ParsedData Parse()
+        {
+            return new ParsedData
+            {
+                LocationIdentifier = _csvParser.GetRequiredStringByLabel(AquaCalcConstants.GageId),
```
Comment (TheRealDougSchmidt) [inline src/AquaCalc5000/Parsers/HeaderParser.cs:19]: We'll need to add a configurable behaviour here. This is my fault. I don't think I spec'd that well enough in the JIRA issue.

The plugin needs a configuration setting, named something like "AssumeUsgsSiteIdentifiers", which should default to true (so Nebraska won't need to change the default behaviour). Other consumers may want to disable the setting.

Nebraska has historically used site identifiers that follow the USGS naming convention, but omit the leading zeros. But their NG system is now using the full 8-digit identifier, for shared sites.

So these AquaCalc files from Nebraska contain `GAGE ID#  ,6690500` but must be mapped to `06690500` (inserting one leading zero) before asking the framework to look up the location.

If the plugin calls `IFieldDataAppender.GetLocationByIdentifier("6690500")`, the framework halts all execution. It does not throw an exception (which the plugin could theoretically catch), but instead the framework just shuts down the appdomain sandbox. The call to GetLocationByIdentifier NEVER RETURNS. It's so gross!

So we'll need some optional padding logic, like this:
```C#
private string ConvertGageIdToLocationIdentifier(string gageId)
{
    if (!Config.AssumeUsgsSiteIdentifiers)
        return gageId;

    return int.TryParse(gageId, out var numericId)
        ? $"{numericId:D8}"
        : gageId;
}
```

The configuration that stores this setting is also fussy. I'd recommend using a JSON config file pattern, like the SxS plugin uses. https://github.com/AquaticInformatics/sxs-pro-field-data-plugin/blob/master/src/SxSPro/Config.cs

This technique will include a Config.json file in the plugin manifest, with the setting defaulting to true,

And someone else can manually tweak the `%ProgramData%\Aquatic Informatics\AQUARIUS Server\FieldDataPlugins\AquaCalc5000\Config.json` file to their needs.

File: src/AquaCalc5000/Parsers/HeaderParser.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                StartDate = GetVisitDate(),
+                Transect = _csvParser.GetRequiredIntByLabel(AquaCalcConstants.Transect),
+                UserId = _csvParser.GetRequiredStringByLabel(AquaCalcConstants.UserId),
+
```
Comment (TheRealDougSchmidt) [inline src/AquaCalc5000/Parsers/HeaderParser.cs:23]: We should also parse out the `AQCUSH8c` firmware version from the CSV header.

```
AquaCalc 5000 (tm)  by JBS Instruments 
Firmware Version AQCUSH8c (c)1995-2000
```

File: src/AquaCalc5000/Parsers/ParsedData.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace AquaCalc5000.Parsers
+{
+    public class ParsedData
+    {
+        public string LocationIdentifier { get; set; }
+        public DateTime StartDate { get; set; }
+        public int Transect { get; set; }
+        public string UserId { get; set; }
+
```
Comment (TheRealDougSchmidt) [inline src/AquaCalc5000/Parsers/ParsedData.cs:12]: I think we will also need to parse the meter calibration equations from the CSV file.

```
      METER TYPE  ,Price AA 1:1 ST2
 METER CONST. C1  ,2.2048
 METER CONST. C2  ,0.0178
 METER CONST. C3  ,2.2048
 METER CONST. C4  ,0.0178
 METER CONST. C5  ,0.0
```

**Review by TheRealDougSchmidt — APPROVED** [review]
This looks really good so far. I think there are a few more additions needed, but I'm going to go ahead and merge this, so that you can make some incremental additions with the missing features.

**Comment by BillChen-AI** [comment]
Thanks Doug for pointing out the missing bits. I'll work on them shortly and submit a new PR.

**Pull request merged**